lib/_http_agent.js

```diff
@@ -496,10 +497,17 @@ Agent.prototype.keepSocketAlive = function keepSocketAlive(socket) {
       const hint = RegExpPrototypeExec(/^timeout=(\d+)/, keepAliveHint)?.[1];
 
       if (hint) {
-        const serverHintTimeout = NumberParseInt(hint) * 1000;
-
-        if (serverHintTimeout < agentTimeout) {
-          agentTimeout = serverHintTimeout;
+        // Let the timer expires before the announced timeout to reduce
+        // the likelihood of ECONNRESET errors
+        let serverHintTimeout = (NumberParseInt(hint) * 1000) - 1000;
+        serverHintTimeout = serverHintTimeout > 0 ? serverHintTimeout : 0;
+        if (serverHintTimeout === 0) {
+          // cannot safely reuse the socket because the server timeout is too short
+          canKeepSocketAlive = false;
```

Comment:
I have updated the PR to just skip keep alive if the timeout is too short. In my local build the test now pass.